### PR TITLE
[Component with its own class name as service type ] LPS-194013 Refactor Registry classes from Dynamic Data Mapping module

### PR DESCRIPTION
**Pull request by GabrielHenriquedev**

### CODE REVIEW

**Comment by GabrielHenriquedev** [comment]
ci:test:sf

**Review by MrJohn1911 — APPROVED** [review]

**Comment by GabrielHenriquedev** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:data-engine

**Comment by catarineaguiar** [comment]
@GabrielHenriquedev semver error:

```
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/dynamic-data-mapping/dynamic-data-mapping-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.dynamic.data.mapping.util              MAJOR      8.0.0      8.0.0      9.0.0      VERSION INCREASE REQUIRED
     [exec]     	-   class      com.liferay.dynamic.data.mapping.util.DDMDisplayRegistry
     [exec]     		-   method     <init>()
     [exec]     			-   return     void
     [exec]     		-   method     activate(org.osgi.framework.BundleContext)
     [exec]     			-   access     protected
     [exec]     			-   annotated  org.osgi.service.component.annotations.Activate
     [exec]     			-   return     void
     [exec]     		-   method     clone()
     [exec]     			-   access     protected
     [exec]     			-   return     java.lang.Object
     [exec]     		-   method     deactivate()
     [exec]     			-   access     protected
     [exec]     			-   annotated  org.osgi.service.component.annotations.Deactivate
     [exec]     			-   return     void
     [exec]     		-   method     equals(java.lang.Object)
     [exec]     			-   return     boolean
     [exec]     		-   method     finalize()
     [exec]     			-   access     protected
     [exec]     			-   return     void
     [exec]     		-   method     getDDMDisplay(java.lang.String)
     [exec]     			-   return     com.liferay.dynamic.data.mapping.util.DDMDisplay
     [exec]     		-   method     getPortletIds()
     [exec]     			-   return     java.lang.String[]
     [exec]     		-   method     hashCode()
     [exec]     			-   return     int
     [exec]     		-   method     toString()
     [exec]     			-   return     java.lang.String
     [exec]     		-   annotated  org.osgi.service.component.annotations.Component
     [exec]     			-   property   service.0=com.liferay.dynamic.data.mapping.util.DDMDisplayRegistry
     [exec]     	+   class      com.liferay.dynamic.data.mapping.util.DDMDisplayRegistryUtil
     [exec]     		+   method     <init>()
     [exec]     			+   return     void
     [exec]     		+   method     clone()
     [exec]     			+   access     protected
     [exec]     			+   return     java.lang.Object
     [exec]     		+   method     equals(java.lang.Object)
     [exec]     			+   return     boolean
     [exec]     		+   method     finalize()
     [exec]     			+   access     protected
     [exec]     			+   return     void
     [exec]     		+   method     getDDMDisplay(java.lang.String)
     [exec]     			+   access     static
     [exec]     			+   return     com.liferay.dynamic.data.mapping.util.DDMDisplay
     [exec]     		+   method     getPortletIds()
     [exec]     			+   access     static
     [exec]     			+   return     java.lang.String[]
     [exec]     		+   method     hashCode()
     [exec]     			+   return     int
     [exec]     		+   method     toString()
     [exec]     			+   return     java.lang.String
     [exec]     [Baseline Warning] Bundle Version Change Recommended: 26.0.0
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-08/test-1-39/test-portal-acceptance-pullrequest(master)/6967/modules-semantic-versioning-jdk8/0/1/jenkins-console.txt.gz?authuser=0)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/util/DDMDisplayRegistryUtil.java [hunk 1]
```diff
@@ -42,11 +46,4 @@ protected void activate(BundleContext bundleContext) {
 					ddmDisplay.getPortletId())));
```
Comment (shuyangzhou) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/util/DDMDisplayRegistryUtil.java:46]: @GabrielHenriquedev please redo this part by property filtering. I think all DDMDisplay service has the "javax.portlet.name" property, if some one is missing it, add it.

Most DDMDisplay are components, we prefer not to pull up the service to trigger component activation during collecting.

CC @rafaprax please watch out for this kind of things in future pulls, I am planning to make some global changes to delay component based services activation system wide.

**Comment by shuyangzhou** [comment]
@GabrielHenriquedev See the comment above, please resend with the fix, thanks!

**Pull request closed without merging**